controllers/single_page/collections.php

```diff
@@ -45,15 +45,11 @@ public function submit()
             if (!$collection || !$collection->is('collection')) {
                 $this->error->add(t('Invalid collection object.'));
                 unset($collection);
+            } else {
+                $validator->validate($form, $collection);
             }
         }
 
-        if ($collection) {
-            $validator->validate($form, ProcessorInterface::REQUEST_TYPE_UPDATE);
-        } else {
-            $validator->validate($form, ProcessorInterface::REQUEST_TYPE_ADD);
-        }
-
```

Comment:
This change makes it so that validation doesn't run for added entries. Instead maybe change:

- `$collection = false` to `$collection = null`
- `unset($collection)` to `$collection = null`
- `if ($collection) {` conditional to:
```
$validator->validate($form, $collection);
```

That way we always run validation, and when collection isn't successfully resolved we pass null